clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -2877,10 +2877,144 @@ void CodeGenFunction::EmitMultiVersionResolver(
   case llvm::Triple::aarch64:
     EmitAArch64MultiVersionResolver(Resolver, Options);
     return;
+  case llvm::Triple::riscv32:
+  case llvm::Triple::riscv64:
+    EmitRISCVMultiVersionResolver(Resolver, Options);
+    return;
 
   default:
-    assert(false && "Only implemented for x86 and AArch64 targets");
+    assert(false && "Only implemented for x86, AArch64 and RISC-V targets");
+  }
+}
+
+static int getPrioiryFromAttrString(StringRef AttrStr) {
+  SmallVector<StringRef, 8> Attrs;
+
+  AttrStr.split(Attrs, ";");
+
+  // Default Pri is zero.
+  int Pri = 0;
+  for (auto Attr : Attrs) {
+    if (Attr.starts_with("priority=")) {
+      Attr.consume_front("priority=");
```

Comment:
consume_front returns a bool if it consumes the string so I think you can write

```
if (Attr.consume_front("priority=")) {
    int Result;
    if (!Attr.getAsInteger(0, Result)) {
        Pri = **Result;
}
```

Reply:
Replace starts_with with consume_front here. 

By the way, does the '**' in 'Pri = **Result' have a specific meaning? Or just typo?